src/lib/core/style/_checkbox-common.scss

```diff
@@ -1,7 +1,7 @@
 @import './variables';
 
 // The width/height of the checkbox element.
-$mat-checkbox-size: $mat-toggle-size !default;
+$mat-checkbox-size: 16px !default;
```

Comment:
If the checkbox is getting smaller, I think we need to add an amount of clickable padding around it such that the total size is the same. Two reasons:
* Layouts may shift, which will be very hard to merge
* Our click targets for toggles are already too small as it is (#8000)

Reply:
As per our discussion on Slack:
* Screenshot diffs will change regardless since the square is smaller. As for layouts changing, it might not be that big of a difference, because we don't have line wrapping on `mat-checkbox`.
* Click targets shouldn't be an issue, because checkboxes have a large label next to them that is also clickable.

I think that we should do a presubmit to see how big of an issue this would be.